tests/cache/models/subjects/getAll.js

```diff
@@ -0,0 +1,279 @@
+      done();
+    });
+  });
+
+  it('sort option asc', (done) => {
+    api.get(`${path}?sort=absolutePath`)
+    .set('Authorization', token)
+    .expect(constants.httpStatus.OK)
+    .end((err, res) => {
+      if (err) {
+        done(err);
+      }
+
+      expect(res.body.length).to.be.equal(3);
+      expect(res.body[0].absolutePath).to.be.equal(subject1);
+      expect(res.body[1].absolutePath).to.be.equal(subject2);
+      expect(res.body[2].absolutePath).to.be.equal(subject3);
+      done();
+    });
+  });
+
+  it('sort option desc', (done) => {
+    api.get(`${path}?sort=-absolutePath`)
+    .set('Authorization', token)
+    .expect(constants.httpStatus.OK)
+    .end((err, res) => {
+      if (err) {
+        done(err);
+      }
+
+      expect(res.body.length).to.be.equal(3);
+      expect(res.body[0].absolutePath).to.be.equal(subject3);
+      expect(res.body[1].absolutePath).to.be.equal(subject2);
+      expect(res.body[2].absolutePath).to.be.equal(subject1);
+      done();
+    });
+  });
+
+  it('fields filter returns in the asc order', (done) => {
+    api.get(`${path}?fields=name`)
+    .set('Authorization', token)
+    .expect(constants.httpStatus.OK)
+    .end((err, res) => {
+      if (err) {
+        done(err);
+      }
+
+      expect(res.body.length).to.be.equal(3);
+      expect(res.body[0].name).to.be.equal(subject1);
+
+      // name is in absolutePath
+      expect(res.body[1].name).to.be.equal(subject2.substr(subject1.length + 1));
+      expect(res.body[2].name).to.be.equal(subject3.substr(subject1.length + 1));
+      done();
+    });
+  });
+
+  it('fields filter returns expected number of keys', (done) => {
+    api.get(`${path}?fields=name`)
+    .set('Authorization', token)
+    .expect(constants.httpStatus.OK)
+    .end((err, res) => {
+      if (err) {
+        done(err);
+      }
+
+      expect(res.body.length).to.be.equal(3);
+      expect(Object.keys(res.body[0]).length).to.equal(3);
+      expect(Object.keys(res.body[1])).to.contain('id', 'name', 'apiLinks');
+      done();
+    });
+  });
+
+  it('fields filter returns apiLinks', (done) => {
+    api.get(`${path}?fields=name`)
+    .set('Authorization', token)
+    .expect(constants.httpStatus.OK)
+    .end((err, res) => {
+      if (err) {
+        done(err);
+      }
+
+      expect(res.body.length).to.be.equal(3);
+      expect(Array.isArray(res.body[0].apiLinks)).to.be.true;
+      expect(Array.isArray(res.body[1].apiLinks)).to.be.true;
+      expect(Array.isArray(res.body[2].apiLinks)).to.be.true;
+      done();
+    });
+  });
+
+  it('limit filter', (done) => {
+    api.get(`${path}?limit=1`)
+    .set('Authorization', token)
+    .expect(constants.httpStatus.OK)
+    .end((err, res) => {
+      if (err) {
+        done(err);
+      }
+
+      expect(res.body.length).to.be.equal(1);
+      expect(res.body[0].name).to.be.equal(subject1);
+      done();
+    });
+  });
+
+  it('offset filter', (done) => {
```

Comment:
I think the api links changes when we apply offset filter, Can you please check?